src/threadpool.c

```diff
@@ -311,19 +311,33 @@ void uv__work_done(uv_async_t* handle) {
   QUEUE* q;
   QUEUE wq;
   int err;
+  int nevents;
 
   loop = container_of(handle, uv_loop_t, wq_async);
   uv_mutex_lock(&loop->wq_mutex);
   QUEUE_MOVE(&loop->wq, &wq);
   uv_mutex_unlock(&loop->wq_mutex);
 
+  nevents = 0;
+
   while (!QUEUE_EMPTY(&wq)) {
     q = QUEUE_HEAD(&wq);
     QUEUE_REMOVE(q);
 
     w = container_of(q, struct uv__work, wq);
     err = (w->work == uv__cancelled) ? UV_ECANCELED : 0;
     w->done(w, err);
+    nevents++;
+  }
+
+  /* If no done() callbacks were called, then no need to increment events.
+   * Since we're also not doing a correction of -1 then the call to this fn
+   * will still count as an event. */
+  if (nevents > 0) {
```

Comment:
```suggestion
  if (nevents > 1) {
```
Because this block is a no-op when `nevents == 1`, I think?

The preceding comment is kind of confusing, in particular "also not doing a correction of -1". It's doing that correction, isn't it?

Reply:
Reworded the comment.